source/DG.Core/Scanners/AssemblyScanner.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace DG.Core.Scanners
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    public class AssemblyScanner : IAssemblyScanner
+    {
+        public IEnumerable<Type> GetAssemblyTypes(string assemblyPath)
+        {
+            Assembly assembly = Assembly.LoadFrom(assemblyPath);
+            return assembly.GetTypes();
+        }
+
+        public Type GetType(string assemblyPath, string namespaceName, string typeName)
+        {
+            Assembly assembly = Assembly.LoadFrom(assemblyPath);
+            return assembly.GetType(namespaceName + "." + typeName);
```

Comment:
We can create new variable 
`private char nameAndTypeSeparator = '.';` 
and use it here 
`assembly.GetType(namespaceName + nameAndTypeSeparator + typeName)`

Reply:
what if we use extensions here?

Reply:
I have reworked this part